src/ML-UF3/pair_uf3.cpp

```diff
@@ -0,0 +1,1313 @@
+  n2b_coeff.resize(num_of_elements + 1);
+  UFBS2b.resize(num_of_elements + 1);
+  for (int i = 1; i < num_of_elements + 1; i++) {
+    n2b_knot[i].resize(num_of_elements + 1);
+    n2b_coeff[i].resize(num_of_elements + 1);
+    UFBS2b[i].resize(num_of_elements + 1);
+  }
+  if (pot_3b) {
+    // Contains info about wether UF potential were found for type i, j and k
+    memory->create(setflag_3b, num_of_elements + 1, num_of_elements + 1, num_of_elements + 1,
+                   "pair:setflag_3b");
+    // Contains info about 3-body cutoff distance for type i, j and k
+    memory->create(cut_3b, num_of_elements + 1, num_of_elements + 1, num_of_elements + 1,
+                   "pair:cut_3b");
+    // Contains info about 3-body cutoff distance for type i, j and k
+    // for constructing 3-body list
+    memory->create(cut_3b_list, num_of_elements + 1, num_of_elements + 1, "pair:cut_3b_list");
+    // Contains info about minimum 3-body cutoff distance for type i, j and k
+    memory->create(min_cut_3b, num_of_elements + 1, num_of_elements + 1, num_of_elements + 1, 3,
+                    "pair:min_cut_3b");
+  //Contains info about type of knot_spacing--> 0 = uniform knot spacing (default)
+  //1 = non-uniform knot spacing
+  memory->create(knot_spacing_type_3b, num_of_elements + 1, num_of_elements + 1,
+          num_of_elements + 1, "pair:knot_spacing_3b");
+
+
+    // setting cut_3b and setflag = 0
+    for (int i = 1; i < num_of_elements + 1; i++) {
+      for (int j = 1; j < num_of_elements + 1; j++) {
+        cut_3b_list[i][j] = 0;
+        for (int k = 1; k < num_of_elements + 1; k++) {
+          cut_3b[i][j][k] = 0;
+          min_cut_3b[i][j][k][0] = 0;
+          min_cut_3b[i][j][k][1] = 0;
+          min_cut_3b[i][j][k][2] = 0;
+        }
+      }
+    }
+    n3b_knot_matrix.resize(num_of_elements + 1);
+    UFBS3b.resize(num_of_elements + 1);
+    for (int i = 1; i < num_of_elements + 1; i++) {
+      n3b_knot_matrix[i].resize(num_of_elements + 1);
+      UFBS3b[i].resize(num_of_elements + 1);
+      for (int j = 1; j < num_of_elements + 1; j++) {
+        n3b_knot_matrix[i][j].resize(num_of_elements + 1);
+        UFBS3b[i][j].resize(num_of_elements + 1);
+      }
+    }
+    memory->create(neighshort, maxshort, "pair:neighshort");
+  }
+}
+
+void PairUF3::uf3_read_pot_file(int itype, int jtype, char *potf_name)
+{
+  utils::logmesg(lmp, "UF3: {} file should contain UF3 potential for {} {}\n", \
+          potf_name, itype, jtype);
+
+  if (!platform::file_is_readable(potf_name))
+    error->all(FLERR, "UF3: {} file is not readable", potf_name);
+
+  FILE *fp;
+  fp = utils::open_potential(potf_name, lmp, nullptr);
+
+  TextFileReader txtfilereader(fp, "UF3:POTFP");
+  txtfilereader.ignore_comments = false;
+
+  std::string temp_line = txtfilereader.next_line(1);
+  Tokenizer file_header(temp_line);
+
+  if (file_header.count() != 2)
+    error->all(FLERR, "UF3: Expected only two words on 1st line of {} but found \n\
```

Comment:
The trailing backslash is rather unusual. Is fixed in my PR branch. also, there don't need to be so many newlines.